src/components/searchWidget.js

```diff
@@ -59,13 +77,13 @@ export default class searchWidget extends Component {
     }
 
     displayResults(r){
-        return <li href={r.repoUrl}> <ul> <li>{r.name}</li><li>{r.stars}</li><li>{r.language}</li><li>{r.topics}</li></ul></li>
+        return <li href={r.repoUrl} key={r.name}> <ul> <li>{r.name}</li><li>{r.stars}</li><li>{r.language}</li><li>{r.topics}</li></ul></li>
     }
 
     render(){
             return (
             <ul>
-                {this.repos.map(this.displayResults(r))}
+                {Array.isArray(this.repos) && this.repos.map((r) => this.displayResults(r))}
```

Comment:
What's the scenario when Array.isArray(this.repos) returns false here? Is that a failure scenario which we should log or throw exception, for visibility to troubleshoot?

Reply:
The Array.isArray check is essentially an override for the testing of this component. The map function requires that the input is an array, though due to the way the testing framework works, this.repos is undefined in most of the tests.  This should never be the case in actual execution.